cluster-migration/src/main/java/com/wipro/ats/bdre/clustermigration/MigrationPreprocessor.java

```diff
@@ -0,0 +1,403 @@
+        //TODO: if previous list is empty it means there are no log entries in process log currently. But this could be the first run of the whole migration program, hence there will obviously be no log entries, in this case add logic to test destination columns using gethivejdbcconnection or you can skip this logic entirely and get columns always from hive metadata.
+        if(previousColumnList.size()>0) addedColumnList.removeAll(previousColumnList);
+
+        for (String addedColumn : addedColumnList) {
+            LOGGER.debug("addedColumn = " + addedColumn);
+        }
+        return addedColumnList;
+    }
+
+    private void logCurrentSourcePartitions(List<String> sourcePartitionList, String processId, String instanceExecId, String table) {
+        ProcessLog processLog = new ProcessLog();
+        List<ProcessLogInfo> partitionLogInfoList = new ArrayList<>();
+        for (String sourcePartition : sourcePartitionList) {
+            ProcessLogInfo processLogInfo = new ProcessLogInfo();
+            processLogInfo.setAddTs(new Date());
+            processLogInfo.setProcessId(Integer.parseInt(processId));
+            processLogInfo.setMessage(sourcePartition);
+            processLogInfo.setInstanceRef(Long.parseLong(instanceExecId));
+            processLogInfo.setLogCategory("C2C");
+            processLogInfo.setMessageId(table + " partition");
+            partitionLogInfoList.add(processLogInfo);
+        }
+        processLog.logList(partitionLogInfoList);
+    }
+
+    private void logCurrentSourceColumns(List<String> sourceColumnList, String processId, String instanceExecId, String table) {
+        ProcessLog processLog = new ProcessLog();
+        List<ProcessLogInfo> columnLogInfoList = new ArrayList<>();
+        for (String sourceColumn : sourceColumnList) {
+            ProcessLogInfo processLogInfo = new ProcessLogInfo();
+            processLogInfo.setAddTs(new Date());
+            processLogInfo.setProcessId(Integer.parseInt(processId));
+            processLogInfo.setMessage(sourceColumn);
+            processLogInfo.setInstanceRef(Long.parseLong(instanceExecId));
+            processLogInfo.setLogCategory("C2C");
+            processLogInfo.setMessageId(table + " column");
+            columnLogInfoList.add(processLogInfo);
+        }
+        processLog.logList(columnLogInfoList);
+    }
+
+    private void formStageAndDestTableDDLs(Statement st, List<String> sourceColumnList, String sourceDb, String destDb, String table, String sourceStgtable, String bdreTechPartition, String processId,String instanceExecId) throws Exception {
+        ResultSet rsPartitionList = st.executeQuery("desc "+table);
+        int index = 0;
+        StringBuffer partitionList = new StringBuffer("");
+        List<String> sourcePartitionColumnList = new ArrayList<>();
+        while (rsPartitionList.next()) {
+            if (rsPartitionList.getString(1).equals("# Partition Information")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "# Partition Information" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1132)